src/index.ts

```diff
@@ -58,55 +57,53 @@ export async function createWebdriverIO(opts: ProgramOpts) {
+        process.argv[1].includes(`${path.sep}${pm}${path.sep}`) ||
+        // for NPM and Yarn check for "~/.npm/npx/..." or "~/.yarn/bin/create-wdio"
+        process.argv[1].includes(`${path.sep}.${pm}${path.sep}`)
+    )) || 'npm'
+
+    const hasPackageJson = await fs.access(path.resolve(root, 'package.json')).then(() => true).catch(() => false)
+    if (!hasPackageJson) {
         await fs.mkdir(root, { recursive: true })
         await fs.writeFile(
             path.resolve(root, 'package.json'),
             JSON.stringify({
                 name: 'my-new-project',
-                type: 'module'
+                type: 'module',
             }, null, 2)
         )
-
-        /**
-         * find package manager that was used to create project
-         */
-        const pm = PMs.find((pm) => (
-            // for pnpm check for "~/Library/pnpm/store/v3/..."
-            process.argv[1].includes(`${path.sep}${pm}${path.sep}`) ||
-            // for NPM and Yarn check for "~/.npm/npx/..." or "~/.yarn/bin/create-wdio"
-            process.argv[1].includes(`${path.sep}.${pm}${path.sep}`)
-        )) || 'npm'
-
-        /**
-         * create a package-lock.json, yarn.lock or pnpm-lock.yaml so
-         * that `detect-package-manager` doesn't mark it as a default
-         * Yarn project
-         * @see https://github.com/egoist/detect-package-manager/issues/11
-         */
-        await runProgram(pm, ['install'], { cwd: root, stdio: 'ignore' })
     }
-
-    console.log(`\nInstalling ${chalk.bold('@wdio/cli')} to initialize project...`)
-    const pm = await detect({ cwd: root })
-    const args = [INSTALL_COMMAND[pm]]
-    if (pm === 'yarn') {
-        args.push('--exact', '--cwd', root)
+    if (!opts.skipCLI) {
+        console.log(`\nInstalling ${chalk.bold('@wdio/cli')} to initialize project...`)
+        const args = [INSTALL_COMMAND[pm]]
+        if (pm === 'yarn') {
+            args.push('--exact', '--cwd', root)
+        }
+        if (opts.dev) {
+            args.push(DEV_FLAG[pm])
+        }
+        args.push(`@wdio/cli${npmTag}`)
+        await runProgram(pm, args, { cwd: root, stdio: 'ignore' })
+        console.log(chalk.green.bold('✔ Success!'))
     }
-    if (opts.dev) {
-        args.push(DEV_FLAG[pm])
-    }
-    args.push(`@wdio/cli${npmTag}`)
-    await runProgram(pm, args, { cwd: root, stdio: 'ignore' })
-    console.log(chalk.green.bold('✔ Success!'))
 
-    return runProgram('npx', [
-        WDIO_COMMAND,
-        'config',
-        ...(opts.yes ? ['--yes'] : [])
-    ], { cwd: root })
+    if (pm === 'npm') {
+        return runProgram('npx', [
+            WDIO_COMMAND,
+            'config',
+            ...(opts.yes ? ['--yes'] : [])
+        ], { cwd: root })
+    } else {
+        return runProgram(pm, [
+            'run',
+            WDIO_COMMAND,
+            'config',
+            ...(opts.yes ? ['--yes'] : [])
+        ], { cwd: root })
+    }
```

Comment:
Since nothing else is different I would do
```suggestion
return runProgram(pm === 'npm' ? 'npx' : pm, [
    WDIO_COMMAND,
    'config',
    ...(opts.yes ? ['--yes'] : [])
], { cwd: root })
```